bin/debug50

```diff
@@ -20,6 +20,11 @@ if [ "$1" = "-HUP" ]; then
     exit 0
 fi
 
+if ! [ -f "$1" ]; then
+    echo "debug50: $1: No such file"
+    exit 1
```

Comment:
I think this will still result in 

```
Can't debug this program! Afraid debug50 only supports compiled C programs!
```

being printed if `$1` is a dir?

Reply:
Nope, `-f` checks if `"$1"` is a regular file. `-d` checks for directories.

Reply:
But if `$1` is a dir, `debug50` will still print

```
Can't debug this program! Afraid debug50 only supports compiled C programs!
```

which isn't accurate, since `$1` isn't a "program" in that case!